### DISCUSSION TITLE
Saving settings for mesh leveling

### DISCUSSION THREAD

**Discussion by czaras** [Jyers/Marlin#532] [Q&A] [answered]
My question is about this firmware: E3V2-ManualMesh-3x3-v4.2.2.bin
Let's say I already did manual mesh leveling and finished it by pressing Save on printer's panel. Should I also go to Cura printer's Start G-code settings and add the line: 
`M420 S1 ;Load Mesh Bed Level
`
like the guy is [doin right here](https://youtu.be/9TCqQF374GU?t=185)

And secondly: if I'd decide to compile the firmware myself and would like to have this mesh leveling always active, should I uncomment one of these line (and then I don't need to modify my Start G-Code?")?
``//#define RESTORE_LEVELING_AFTER_G28``
`//#define ENABLE_LEVELING_AFTER_G28`

**Comment by aryajuanda** [comment, marked as answer]
if i m not mistaken restore leveling will load saved mesh after G28 (homing) and enable leveling after g28 is do ABL (G29) after G28 (homing) .. CMIIW.. haven't try it before..


Normally G28 leaves leveling disabled on completion. Enable one of
these options to restore the prior leveling state or to always enable
leveling immediately after G28.
#define RESTORE_LEVELING_AFTER_G28 
//#define ENABLE_LEVELING_AFTER_G28



because i use bltouch i use 2 kind of profile in slicer, for a small print in the middle i use : 
G28 ; home all axis
M420 S1 ; Use Saved Mesh

and for large print or of i m not sure my bed is level i use : 
G28 ; home all axis
G29 ; ABL

**Reply by Jyers** [reply]
@aryajuanda is correct. Also to be clear you should always enable leveling in your start gcode as restore after g28 will still leave it disabled if it was disabled before homing.

**Comment by czaras** [comment]
Ok, so let's summarize it because I'm not sure if I understand it correctly.
I should add M420 line to Start G-code because after I make leveling on printer and save it, it doesn't mean my printer will use it as it is. It will use it only when I add M420 line to Start G-code, right?

**Reply by Jyers** [reply]
Correct, M420 S1 will ensure that the printer uses the mesh.
